application/src/main/java/org/opentripplanner/graph_builder/module/DirectTransferGenerator.java

```diff
@@ -115,31 +132,48 @@ public void buildGraph() {
             if (sd.stop.transfersNotAllowed()) {
               continue;
             }
-            distinctTransfers.put(
-              new TransferKey(stop, sd.stop, sd.edges),
-              new PathTransfer(stop, sd.stop, sd.distance, sd.edges)
-            );
+            TransferKey transferKey = new TransferKey(stop, sd.stop, sd.edges);
+            PathTransfer pathTransfer = distinctTransfers.get(transferKey);
+            if (pathTransfer == null) {
+              EnumSet<StreetMode> modes = EnumSet.of(mode);
+              distinctTransfers.put(
+                transferKey,
+                new PathTransfer(stop, sd.stop, sd.distance, sd.edges, modes)
+              );
+            } else {
+              pathTransfer.addMode(mode);
+            }
```

Comment:
You could add a comment here to explain what these different cases mean.

Reply:
I added comments for the different cases